### PR TITLE
Improve systemd-analyze security a bit and other assorted bits

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/analyze/analyze-security.c [hunk 1]
```diff
@@ -493,7 +493,7 @@ static int assess_system_call_architectures(
 
 #if HAVE_SECCOMP
 
-static bool syscall_names_in_filter(Set *s, bool allow_list, const SyscallFilterSet *f) {
+static bool syscall_names_in_filter(Set *s, bool allow_list, const SyscallFilterSet *f, const char **ret_offending_syscall) {
         const char *syscall;
```
Comment (poettering) [inline src/analyze/analyze-security.c:497]: i think we should initialize `ret_offending_syscall` when we return false to `NULL`, to follow our usual style

File: src/analyze/analyze-security.c [hunk 2]
```diff
@@ -530,42 +532,48 @@ static int assess_system_call_filter(
...
-
         assert(a);
         assert(info);
         assert(ret_badness);
         assert(ret_description);
 
         assert(a->parameter < _SYSCALL_FILTER_SET_MAX);
-        f = syscall_filter_sets + a->parameter;
+        const SyscallFilterSet *f = syscall_filter_sets + a->parameter;
+
```
Comment (poettering) [inline src/analyze/analyze-security.c:543]: why the newline?

Comment (keszybz) [reply src/analyze/analyze-security.c:543]: The assert makes sure that the `*f` assignment is OK.
The two variable declarations below are a separate thing.

**Comment by keszybz** [comment]
Updated to always initialize `ret_offending_syscall`, no other changes.

**Pull request merged**